packages/data-stores/src/site/actions.ts

```diff
@@ -403,64 +421,11 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 			yield setGlobalStyles( siteSlug, activatedTheme.stylesheet, globalStyles, activatedTheme );
 		}
 
-		return activatedTheme;
-	}
-
-	function* runThemeSetupOnSite(
-		siteSlug: string,
-		selectedDesign: Design,
-		options?: DesignOptions
-	) {
-		const { recipe } = selectedDesign;
-
-		/*
-		 * Anchor themes are set up directly via Headstart on the server side
-		 * so exclude them from theme setup.
-		 */
-		const anchorDesigns = [ 'hannah', 'gilbert', 'riley' ];
-		if ( anchorDesigns.indexOf( selectedDesign.template ) >= 0 ) {
-			return;
-		}
-
-		const themeSetupOptions: ThemeSetupOptions = {
-			trim_content: options?.trimContent ?? true,
-		};
-
-		if ( options?.posts_source_site_id ) {
-			themeSetupOptions.posts_source_site_id = options.posts_source_site_id;
-		}
-
-		if ( recipe?.pattern_ids ) {
-			themeSetupOptions.pattern_ids = recipe?.pattern_ids;
+		if ( shouldRunThemeSetup ) {
+			yield runThemeSetupOnSite( siteSlug );
 		}
```

Comment:
![badge](https://img.shields.io/badge/review-nits-green.svg) `/non-blocker`
Since it's a special case, we might feel more comfortable putting it to the test. ✅

Reply:
Updated!